### PR TITLE
Add scale lock summary metric to observe lock durations & count

### PR DESCRIPTION
**Pull request by pdzwart-atlassian**
- Related to #60 
- Added `node_group_scale_lock_duration` histogram to expose lock durations
- Fixed utility of `node_group_scale_lock`; as is now a gauge and is calculated directly from the lock.
- Added `node_group_scale_lock_check_was_locked` counter to expose how many times the lock was checked and found held.
- Some `Makefile` and `Dockerfile` improvements around dependency management and change detection

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -142,14 +142,34 @@ var (
...
 		},
 		[]string{"node_group"},
 	)
+	// NodeGroupScaleLockDuration indicates how long the nodegroup is locked from scaling
+	NodeGroupScaleLockDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "node_group_scale_lock_duration",
+			Namespace: NAMESPACE,
+			Help: "indicates how long the nodegroup is locked from scaling",
+			Buckets: []float64{1, 2, 3, 5, 8, 13, 21, 34, 55},
```
Comment (awprice) [inline pkg/metrics/metrics.go]: Are these buckets in seconds? We might need to have larger buckets as the scale lock could be present for 5-10 minutes depending on how quickly the cloud provider can bring up the new instances and for them to join the cluster.

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -213,6 +233,8 @@ func init() {
 	prometheus.MustRegister(NodeGroupTaintEvent)
 	prometheus.MustRegister(NodeGroupUntaintEvent)
 	prometheus.MustRegister(NodeGroupScaleLock)
+	prometheus.MustRegister(NodeGroupScaleLockDuration)
+	prometheus.MustRegister(NodeGroupScaleLockCheckWasLocked)
```
Comment (awprice) [inline pkg/metrics/metrics.go]: Are you able to add these new metrics to https://github.com/atlassian/escalator/blob/master/docs/metrics.md

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (pdzwart-atlassian) [reply pkg/metrics/metrics.go]: I'll switch to a summary, given quantile results are wanted to be accurate (i.e. not interpolated via bucket values/frequency). Suggest the metric observations will be low enough to not put undue load on prometheus.

File: pkg/metrics/metrics.go (hunk 2 above)
Comment (awprice) [reply pkg/metrics/metrics.go]: 👍 done

**Pull request merged**